pkg/analysis/duplicatemarkers/analyzer.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package duplicatemarkers
+
+import (
+	"fmt"
+	"go/ast"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+
+	kalerrors "sigs.k8s.io/kube-api-linter/pkg/analysis/errors"
+	"sigs.k8s.io/kube-api-linter/pkg/analysis/helpers/markers"
+)
+
+const (
+	name = "duplicatemarkers"
+)
+
+// Analyzer is the analyzer for the duplicatemarkers package.
+// It checks for duplicate markers on struct fields.
+var Analyzer = &analysis.Analyzer{
+	Name:     name,
+	Doc:      "Check for duplicate markers on struct fields.",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspect.Analyzer, markers.Analyzer},
+}
+
+func run(pass *analysis.Pass) (any, error) {
+	inspect, ok := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
```

Comment:
Why not use the inspector helper that determines which fields should/shouldn't be checked?

Reply:
We want to check not only struct field but also type declaration whether it has duplicated marker.

Reply:
The inspector can now look at type specs too, should we update here?

Reply:
Okay. I'll try it